syncstorage/storage/sql/__init__.py

```diff
@@ -460,6 +462,72 @@ def set_items(self, session, userid, collection, items):
         return self._touch_collection(session, userid, collectionid)
 
     @with_session
+    def create_batch(self, session, userid, collection):
+        """Creates a batch in batch_uploads table"""
+        collectionid = self._get_collection_id(session, collection)
+        batchid = ts2bigint(session.timestamp)
+        params = {
+            "batch": batchid,
+            "userid": userid,
+            "collection": collectionid
+        }
+        session.query("CREATE_BATCH", params)
+        return batchid
+
+    @with_session
+    def valid_batch(self, session, userid, collection, batchid):
+        """Checks to see if the batch ID is valid and still open"""
+        collectionid = self._get_collection_id(session, collection)
+        params = {
+            "batch": batchid,
+            "userid": userid,
+            "collection": collectionid
+        }
+        valid = session.query_scalar("VALID_BATCH", params=params)
+        return valid
```

Comment:
Since the batchid is a timestamp, we should be able to answer `false` to `valid_batch` if called with a very old batch id, outside whatever the expiry time is for a batch.